### ISSUE TITLE
Add support to Datadog Metrics destination for retrying requests.

### ISSUE THREAD

**Issue by tobz** [DataDog/saluki#116]
Currently, the Datadog Metrics destination does not retry requests, which means we can easily drop metrics on the floor if we get temporary errors from the Datadog intake.

We should:

- add support for retries in the I/O loop (like #115, we should use `tower` and `ServiceBuilder::retry` here)
- match any existing retry configurability in the Datadog Agent's forwarder (i.e. add a configuration setting, or settings, that match what can be configured in the Datadog Agent today)

## Notes

Example code in Vector of using `tower::ServiceBuilder` to wrap a `Service` and provide retries: https://github.com/vectordotdev/vector/blob/103240846e93b48213cda50e0025c756d217647c/src/sinks/util/service.rs#L398

We'll end up needing to integrate this with `saluki_io::net::client::http::HttpClient`. The underlying `Client<C, B>` from `hyper_util` _does_ implement `tower::Service`, so that's good... but we'd just need to figure out how we want to wrap things.

Basically, I think it'd be good to keep `HttpClient` as-is, in terms of how we call `HttpClient::send` to send a request, and it's asynchronous and all of that... and so that would sort of imply we need to store the wrapped `Client<C, B>` inside of `HttpClient`, which makes the types a little ugly but isn't insurmountable.

**Comment by tobz** [comment]
Closed by #241.
